ext/intl/spoofchecker/spoofchecker_main.c

```diff
@@ -167,3 +167,42 @@ PHP_METHOD(Spoofchecker, setRestrictionLevel)
 }
 /* }}} */
 #endif
+
+#if U_ICU_VERSION_MAJOR_NUM >= 74
+PHP_METHOD(Spoofchecker, areBidiConfusable)
+{
+	int ret;
+	char *id1, *id2;
+	size_t length1, length2;
+	zend_long direction;
+	zval *error_code = NULL;
+	SPOOFCHECKER_METHOD_INIT_VARS;
+
+	if (FAILURE == zend_parse_parameters(ZEND_NUM_ARGS(), "lss|z", &direction, &id1, &length1,
+										 &id2, &length2, &error_code)) {
+		RETURN_THROWS();
+	}
+
+	SPOOFCHECKER_METHOD_FETCH_OBJECT;
+	if (direction != UBIDI_LTR && direction != UBIDI_RTL) {
+		zend_argument_value_error(1, "must be either SpoofChecker::UBIDI_LTR or SpoofChecker::UBIDI_RTL");
+		RETURN_THROWS();
+	}
+	if(length1 > INT32_MAX || length2 > INT32_MAX) {
+		SPOOFCHECKER_ERROR_CODE(co) = U_BUFFER_OVERFLOW_ERROR;
+	} else {
+		ret = uspoof_areBidiConfusableUTF8(co->uspoof, (UBiDiDirection)direction, id1, (int32_t)length1, id2, (int32_t)length2, SPOOFCHECKER_ERROR_CODE_P(co));
+	}
+	if (U_FAILURE(SPOOFCHECKER_ERROR_CODE(co))) {
+		php_error_docref(NULL, E_WARNING, "(%d) %s", SPOOFCHECKER_ERROR_CODE(co), u_errorName(SPOOFCHECKER_ERROR_CODE(co)));
+		RETURN_TRUE;
```

Comment:
So we return true in case on failures? Is this behavior OK? Shouldn't we rather throw?

Reply:
was being consistent with the existing areConfusable method. I may adjust spoofchecker class behavior in a separate commit like I m doing with IntlTimeZone

Reply:
If the failures are related to programmatic errors, then it's surely better to throw. The one error case I can immediately see is however related to string length, which seems like a user-related one, but I guess it should be quite rare to exceed the limit. that's why I'm not really sure what to do. If we threw an exception, then the ugly reference param could also be avoided, which would be a nice advantage.

cc. @Girgias Do you have any opinion about this?

Reply:
Ehhhh not really, but shouldn't this go through the "normal" intl extension error handling mechanism so user can set if they want it to be silent/warning/error?